Orange/data/sql/backend/mssql.py

```diff
@@ -117,17 +117,26 @@ def _guess_variable(self, field_name, field_metadata, inspect_table):
     EST_ROWS_RE = re.compile(r'StatementEstRows="(\d+)"')
 
     def count_approx(self, query):
-        try:
-            with self.connection.cursor() as cur:
+        with self.connection.cursor() as cur:
+            try:
                 cur.execute("SET SHOWPLAN_XML ON")
                 try:
                     cur.execute(query)
                     result = cur.fetchone()
                     return int(self.EST_ROWS_RE.search(result[0]).group(1))
+                except AttributeError:
+                # This is to catch a float received in StatementEstRows
+                # a float is received when the server's statistics are out of date.
+                    pass
                 finally:
                     cur.execute("SET SHOWPLAN_XML OFF")
-        except pymssql.Error as ex:
-            if "SHOWPLAN permission denied" in str(ex):
-                warnings.warn("SHOWPLAN permission denied, count approximates will not be used")
-                return None
-            raise BackendError(str(ex)) from ex
+            except pymssql.Error as ex:
+                if "SHOWPLAN permission denied" in str(ex):
+                    warnings.warn("SHOWPLAN permission denied, count approximates will not be used")
+                    return None
+                raise BackendError(str(ex)) from ex
+            # In case of an AttributeError, give a second chance:
+            # Use the long method for counting
+            cur.execute("SELECT count(*) FROM ( {} ) x".format(query))
```

Comment:
This should not be necessary. When count_approx returns None, an exact count is already performed (see [SqlTable.count_approx](https://github.com/biolab/orange3/blob/master/Orange/data/sql/table.py#L274))